lib/server.js

```diff
@@ -79,16 +80,22 @@ async function initConfiguration(rawConfig) {
     config.repoRoot = path.resolve(config.repoRoot);
     await fse.ensureDir(config.repoRoot);
 
-    // configure logger
-    config.logs = config.logs || {};
-    config.logs.logsDir = path.normalize(config.logs.logsDir || 'logs');
-    await fse.ensureDir(config.logs.logsDir);
-    rootLogger.loggers.set( // Using a uuid in the name here makes collisions extremely unlikely
-      'git-server-errors-6ae5f55e-dbb3-46a0-a596-c238e713c1cc',
-      new FileLogger(path.join(config.logs.logsDir, 'error.log'), { level: config.logs.level || 'info' }),
-    );
-
-    debug(`configuration successfully read: ${config.configPath}`);
+    if (!config.logger) {
+      // configure logger
+      config.logs = config.logs || {};
+      config.logs.logsDir = path.normalize(config.logs.logsDir || 'logs');
+
+      await fse.ensureDir(config.logs.logsDir);
+      rootLogger.loggers.set( // Using a uuid in the name here makes collisions extremely unlikely
+        'git-server-errors-6ae5f55e-dbb3-46a0-a596-c238e713c1cc',
+        new FileLogger(path.resolve(config.logs.logsDir, 'error.log')),
```

Comment:
because the level is already dictated by the `SimpleInterface`. i.e. the filelogger should not decided if it wants to log or not, but be merely a stream.